### ISSUE TITLE
Naming/RescuedExceptionsVariableName autocorrect uses non-descriptive variable name

### ISSUE THREAD

**Issue by jcoyne** [rubocop/rubocop#12042]
Using rubocop 1.54.1 I encountered this warning:
```
lib/traject/writers/solr_better_json_writer.rb:82:37: Naming/RescuedExceptionsVariableName: Use e instead of exception.
    rescue *skippable_exceptions => exception
                                    ^^^^^^^^^
```

And then when you autocorrect it, it uses `e` rather than something more descriptive. 
```
-    if exception || resp.status != 200
-      msg = if exception
-              Traject::Util.exception_to_log_message(exception)
+    if e || resp.status != 200
+      msg = if e
+              Traject::Util.exception_to_log_message(e)
             else
               "Solr error response: #{resp.status}: #{resp.body}"
             end
       logger.error "Could not add record #{c.record_inspect}: #{msg}"
-      logger.debug("\t" + exception.backtrace.join("\n\t")) if exception
+      logger.debug("\t" + e.backtrace.join("\n\t")) if e
```

I think that using the more descriptive variable name in this case results in more readable code.  I think that most practitioners would agree that this change ☝️ is not ideal.

**Comment by koic** [comment]
After some research, `e` was made the default.
https://github.com/rubocop/rubocop/pull/6460#discussion_r232705359

The default can be changed with the `PreferredName: exception` setting in user's .rubocop.yml:
https://docs.rubocop.org/rubocop/1.54/cops_naming.html#configurable-attributes-13

Thank you.

**Comment by jcoyne** [comment]
I think it's worth considering a different behavior when the exception is used only inside the rescue block and when it's used extensively outside the block. I think this is required to have to consistency with related Naming cops such as Naming/MethodParameterName (MinNameLength)

**Comment by bbatsov** [comment]
@jcoyne That would add a lot of complexity to the cop for something that's rarely done in practice. I can't recall the last time I did something with an exception outside of the `rescue` block.
